masu/processor/report_processor.py

```diff
@@ -0,0 +1,480 @@
+        self._cursor_pos = cursor_pos
+        self._compression = compression
+        self._report_name = path.basename(report_path)
+        self._datetime_format = Config.AWS_DATETIME_STR_FORMAT
+        self._batch_size = Config.REPORT_PROCESSING_BATCH_SIZE
+
+        self.processed_report = ProcessedReport()
+
+        # Gather database accessors
+        self.report_common_db = ReportingCommonDBAccessor()
+        self.column_map = self.report_common_db.column_map
+
+        self.report_db = ReportDBAccessor(schema=self._schema_name,
+                                          column_map=self.column_map)
+        self.report_schema = self.report_db.report_schema
+
+        self.current_bill = self.report_db.get_current_cost_entry_bill()
+        self.existing_cost_entry_map = self.report_db.get_cost_entries()
+        self.existing_product_map = self.report_db.get_products()
+        self.existing_pricing_map = self.report_db.get_pricing()
+        self.existing_reservation_map = self.report_db.get_reservations()
+
+        LOG.info('Initialized report processor for file: %s and schema: %s',
+                 self._report_name, self._schema_name)
+
+    def process(self):
+        """Process CUR file.
+
+        Returns:
+            (int): An updated cursor position.
+
+        """
+        row_count = 0
+        bill_id = None
+        opener, mode = self._get_file_opener(self._compression)
+
+        # pylint: disable=invalid-name
+        with opener(self._report_path, mode) as f:
+            reader = csv.DictReader(f)
+            for row in islice(reader, self._cursor_pos, None):
+                if bill_id is None:
+                    bill_id = self._create_cost_entry_bill(row)
+
+                cost_entry_id = self._create_cost_entry(row, bill_id)
+                product_id = self._create_cost_entry_product(row)
+                pricing_id = self._create_cost_entry_pricing(row)
+                reservation_id = self._create_cost_entry_reservation(row)
+
+                self._create_cost_entry_line_item(
+                    row,
+                    cost_entry_id,
+                    bill_id,
+                    product_id,
+                    pricing_id,
+                    reservation_id
+                )
+
+                if len(self.processed_report.line_items) >= self._batch_size:
+                    self._save_to_db()
+
+                    LOG.info('Saving report rows %d to %d', row_count,
+                             row_count + len(self.processed_report.line_items))
+                    row_count += len(self.processed_report.line_items)
+
+                    self._update_mappings()
+
+            if self.processed_report.line_items:
+                self._save_to_db()
+                row_count += len(self.processed_report.line_items)
+            self.report_db.close_psycopg2_connection()
+            self.report_db.close_session()
+
+        return self._cursor_pos + row_count
```

Comment:
So the cursor position is the row in the file?  I was under the impression that this would be the byte offset of the file.  I'll update the docstring in the report stats accessor when we begin integration

Reply:
Yes, my thinking was that since these are CSVs we're never not going to process an entire row. Things get a little whacky if you byte offset a file and then try to wrap the csv package's readers around that offset file.